src/main/scala/uncore/tilelink2/RationalCrossing.scala

```diff
@@ -0,0 +1,177 @@
+// See LICENSE.SiFive for license details.
+
+package uncore.tilelink2
+
+import Chisel._
+import chisel3.internal.sourceinfo.SourceInfo
+import config._
+import diplomacy._
+import util._
+
```

Comment:
FYI, you can use this crossing to go from N:M by going via N:(N*M):M. So it truly is a Rational crossing...